waiter/src/waiter/util/client_tools.clj

```diff
@@ -373,11 +373,12 @@
 
 (defn make-request
   ([waiter-url path &
-    {:keys [body client cookies content-type disable-auth form-params headers
+    {:keys [body client cookies content-type disable-auth fold-chunked-response? form-params headers
             method multipart protocol query-params scheme trailers-fn verbose]
      :or {body nil
           cookies []
           disable-auth nil
+          fold-chunked-response? true
```

Comment:
Instead of `fold-chunked-response?` , should we call it `async?` as basically we want to differentiate between synchronous and asynchronous response. And for asynchronous responses we disable folding.